test/ffmpeg-qsv/transcode/transcoder.py

```diff
@@ -120,15 +137,25 @@ def gen_output_opts(self):
     for n, output in enumerate(self.outputs):
       codec = output["codec"]
       mode = output["mode"]
-
+      width = output.get("width", 0)
+      height = output.get("height", 0)
       _, _, ffparms = self.get_requirements_data("encode", codec, mode)
       assert ffparms is not None, "failed to find a suitable encoder"
 
       ffcodec, ext = ffparms
 
       for channel in xrange(output.get("channels", 1)):
         if "hw" == mode and "sw" == self.mode:
-          opts += " -vf hwupload=extra_hw_frames=64,format=qsv"
+          if width > 0 and height > 0:
+            opts += " -vf 'hwupload=extra_hw_frames=64,format=qsv,vpp_qsv=w={}:h={}'".format(width, height)
```

Comment:
Why hard-coded scaling plugin here and other places?  This is not maintainable code.  Scaling plugin should be obtained from the requirements map.